lib/puppet/provider/zabbix_host/ruby.rb

```diff
@@ -86,6 +102,7 @@ def exists?
 
   def destroy
     zbx.hosts.delete(zbx.hosts.get_id(host: @resource[:hostname]))
+    @property_flush[:destroyed] = true
```

Comment:
awesome PR! I'm always happy to see type/provider adjustments/extensions.
I'm wondering about this change. Does it fix any broken resource cleanup / is it actually related to the TLS implementation?

Reply:
You are right, there is no difference or without this `property_flush`.  In fact, the `destroy` function is not even possible to call in the present state of the module, I had to manually add an `ensure => absent` in the `@@zabbix_host` call in `resources/agent.pp` to test it, and it went flawlessly.

I’ve also removed the `def initialize()` that was only creating this `@property_flush`
```
  def initialize(value = {})
    super(value)
    @property_flush = {}
  end
```

Thanks for pointing this out